### PR TITLE
Leidinggevendenbeheer v2

### PR DESCRIPTION
**Pull request by bfidlers**

### CODE REVIEW

File: app/controllers/leidinggevendenbeheer/bestuursfunctie/contact-info.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+
+export default class LeidinggevendenbeheerBestuursfunctieContactInfoController extends Controller {
+  @service router;
+
+  @tracked showConfirmationDialog = false;
+
+  get isDirty() {
```
Comment (Rahien) [inline app/controllers/leidinggevendenbeheer/bestuursfunctie/contact-info.js:11]: question: do we still use this? i'm guessing all editing is to be done in forms now?

File: app/controllers/leidinggevendenbeheer/bestuursfunctie/functionarissen/edit.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+export default class LeidinggevendenbeheerBestuursfunctieFunctionarissenEditController extends Controller {
+  @service router;
+
+  @tracked initialStatus;
+
+  get statusIsDirty() {
+    return this.initialStatus.get('id') != this.model.status.get('id');
+  }
+
+  get isDirty() {
```
Comment (Rahien) [inline app/controllers/leidinggevendenbeheer/bestuursfunctie/functionarissen/edit.js:15]: same as above

File: app/controllers/leidinggevendenbeheer/bestuursfunctie/functionarissen/new-person.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+
+export default class LeidinggevendenbeheerBestuursfunctieFunctionarissenNewPersonController extends Controller {
```
Comment (Rahien) [inline app/controllers/leidinggevendenbeheer/bestuursfunctie/functionarissen/new-person.js:6]: i think we may not need this anymore as creating persons will be dealt with by the person selector component in the form

File: app/controllers/leidinggevendenbeheer/bestuursfunctie/functionarissen/new.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+
+export default class LeidinggevendenbeheerBestuursfunctieFunctionarissenNewIndexController extends Controller {
+  @service router;
+
+  @action
+  createNewPerson(hasData) {
```
Comment (Rahien) [inline app/controllers/leidinggevendenbeheer/bestuursfunctie/functionarissen/new.js:9]: same, don't need this anymore i think

File: app/router.js [hunk 5]
```diff
@@ -39,6 +39,21 @@ Router.map(function () {
...
 
+  this.route('leidinggevendenbeheer', function () {
+    this.route(
+      'bestuursfunctie',
+      { path: '/:bestuursfunctie_id' },
+      function () {
+        this.route('contact-info');
+        this.route('functionarissen', function () {
+          this.route('edit', { path: '/:functionaris_id/edit' });
+          this.route('new-person');
```
Comment (Rahien) [inline app/router.js:50]: again that route we don't need anymore, just flagging for easy tracking

File: app/routes/leidinggevendenbeheer/bestuursfunctie/functionarissen/new-person.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import Route from '@ember/routing/route';
+
+export default class LeidinggevendenbeheerBestuursfunctieFunctionarissenNewPersonRoute extends Route {
+  resetController(controller, isExiting) {
+    if (isExiting) {
```
Comment (Rahien) [inline app/routes/leidinggevendenbeheer/bestuursfunctie/functionarissen/new-person.js:5]: i think these props are also no longer needed

**Review by Rahien — CHANGES_REQUESTED** [review]

File: app/controllers/leidinggevendenbeheer/bestuursfunctie/contact-info.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+
+export default class LeidinggevendenbeheerBestuursfunctieContactInfoController extends Controller {
+  @service router;
+
+  // TODO before a dialog box was shown when leaving the form with inserted data without saving.
```
Comment (Rahien) [inline app/controllers/leidinggevendenbeheer/bestuursfunctie/contact-info.js:8]: sounds like a good idea! please make a story for it in the backlog!

File: app/controllers/leidinggevendenbeheer/bestuursfunctie/functionarissen/edit.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+
+export default class LeidinggevendenbeheerBestuursfunctieFunctionarissenEditController extends Controller {
+  @service router;
+
+  // TODO make sure filled in data is reset after cancelling
```
Comment (Rahien) [inline app/controllers/leidinggevendenbeheer/bestuursfunctie/functionarissen/edit.js:8]: won't the transition do this automatically? we didn't update any ember models and will fetch the instance again on reloading the route

**Review by Rahien — APPROVED** [review]

File: app/controllers/leidinggevendenbeheer/bestuursfunctie/functionarissen/edit.js (hunk 8 above)
Comment (bfidlers) [reply app/controllers/leidinggevendenbeheer/bestuursfunctie/functionarissen/edit.js:8]: Ah yes with forms this is probably fine, in the legacy code this was partially changed in the model, so this needed to be reset. I'll remove this TODO.

File: app/controllers/leidinggevendenbeheer/bestuursfunctie/contact-info.js (hunk 7 above)
Comment (bfidlers) [reply app/controllers/leidinggevendenbeheer/bestuursfunctie/contact-info.js:8]: done

**Pull request merged**